disnake/ext/commands/params.py

```diff
@@ -63,20 +64,34 @@
 from . import errors
 from .converter import CONVERTER_MAPPING
 
+T_ = TypeVar("T_")
+
 if TYPE_CHECKING:
-    from typing_extensions import Self
+    from typing_extensions import Concatenate, ParamSpec, Self
 
     from disnake.app_commands import Choices
     from disnake.i18n import LocalizationValue, LocalizedOptional
     from disnake.types.interactions import ApplicationCommandOptionChoiceValue
 
+    from .cog import Cog
     from .slash_core import InvokableSlashCommand, SubCommand
 
     AnySlashCommand = Union[InvokableSlashCommand, SubCommand]
 
+    CogT = TypeVar("CogT", bound="Optional[Cog]")
+
+    P = ParamSpec("P")
+
+    InjectionCallback = Union[
+        Callable[Concatenate[CogT, ApplicationCommandInteraction, P], T_],
+        Callable[Concatenate[ApplicationCommandInteraction, P], T_],
+    ]
```

Comment:
appears to be unused

Reply:
my bad

Reply:
Sadly, even though it is not used, pyright complains very much without it /shrug
If keeping it is not a problem, all suggestions are done!

Reply:
Huh, weird. This works fine for me, without pyright complaining:
```diff
diff --git a/disnake/ext/commands/params.py b/disnake/ext/commands/params.py
index 781f14d8..3791599a 100644
--- a/disnake/ext/commands/params.py
+++ b/disnake/ext/commands/params.py
@@ -67,7 +67,7 @@ from .converter import CONVERTER_MAPPING
 T_ = TypeVar("T_")
 
 if TYPE_CHECKING:
-    from typing_extensions import Concatenate, ParamSpec, Self
+    from typing_extensions import ParamSpec, Self
 
     from disnake.app_commands import Choices
     from disnake.i18n import LocalizationValue, LocalizedOptional
@@ -78,15 +78,8 @@ if TYPE_CHECKING:
 
     AnySlashCommand = Union[InvokableSlashCommand, SubCommand]
 
-    CogT = TypeVar("CogT", bound="Optional[Cog]")
-
     P = ParamSpec("P")
 
-    InjectionCallback = Union[
-        Callable[Concatenate[CogT, ApplicationCommandInteraction, P], T_],
-        Callable[Concatenate[ApplicationCommandInteraction, P], T_],
-    ]
-
     from typing_extensions import TypeGuard
 
     TChoice = TypeVar("TChoice", bound=ApplicationCommandOptionChoiceValue)

```